metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1589,17 +1602,28 @@ class MetalsLanguageServer(
       path <- paths.iterator
       if path.isScalaOrJava
     } {
-      indexSourceFile(path, buildTargets.inverseSourceItem(path))
+      indexSourceFile(path, buildTargets.inverseSourceItem(path), None)
     }
   }
 
+  private def toIndexSource(
+      source: AbsolutePath,
+      targetOpt: Option[b.BuildTargetIdentifier]
+  ): AbsolutePath =
+    targetOpt
+      .flatMap(ammonite.generatedScalaPath(_, source))
+      .getOrElse(source)
+
   private def indexSourceFile(
       source: AbsolutePath,
-      sourceItem: Option[AbsolutePath]
+      sourceItem: Option[AbsolutePath],
+      targetOpt: Option[b.BuildTargetIdentifier]
   ): Unit = {
+
     try {
+      val toIndexSource0 = toIndexSource(source, targetOpt)
```

Comment:
I'm not sure it can be moved to `indexWorkspaceSources`, as `indexSourceFile` is called by both `indexWorkspaceSources` and `reindexWorkspaceSources`.

Reply:
I changed `toIndexSource` to `sourceToIndex`, it's clearer this way, yes.